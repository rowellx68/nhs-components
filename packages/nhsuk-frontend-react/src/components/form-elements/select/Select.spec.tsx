import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Default as DefaultStory,
  WithHint as WithHintStory,
  WithError as WithErrorStory,
} from './Select.stories';

const Default = composeStory(DefaultStory, meta);
const WithHint = composeStory(WithHintStory, meta);
const WithError = composeStory(WithErrorStory, meta);

it('should render a select element', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('select')).toBeInTheDocument();
  expect(container.querySelectorAll('option')).toHaveLength(3);
  expect(container).toMatchSnapshot();
});

it('should render a select element with hint', () => {
  const { container } = render(<WithHint />);

  expect(container.querySelector('select')).toBeInTheDocument();
  expect(container.querySelector('.nhsuk-hint')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render a select element with error', () => {
  const { container } = render(<WithError />);

  expect(container.querySelector('select')).toBeInTheDocument();
  expect(container.querySelector('.nhsuk-error-message')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
