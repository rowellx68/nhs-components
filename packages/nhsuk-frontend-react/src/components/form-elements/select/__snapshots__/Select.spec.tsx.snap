// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render a select element 1`] = `
<div>
  <div
    class="nhsuk-form-group"
  >
    <label
      class="nhsuk-label"
      for="select_:r0:"
      id="select_:r0:--label"
    >
      Label text goes here
    </label>
    <select
      aria-labelledby="select_:r0:--label"
      as="select"
      class="nhsuk-select"
      id="select_:r0:"
    >
      <option
        value="option-1"
      >
        Option 1
      </option>
      <option
        value="option-2"
      >
        Option 2
      </option>
      <option
        value="option-3"
      >
        Option 3
      </option>
    </select>
  </div>
</div>
`;

exports[`should render a select element with error 1`] = `
<div>
  <div
    class="nhsuk-form-group nhsuk-form-group--error"
  >
    <label
      class="nhsuk-label"
      for="select_:r4:"
      id="select_:r4:--label"
    >
      Label text goes here
    </label>
    <p
      class="nhsuk-error-message"
      id="select_:r4:--error-message"
    >
      <span
        class="nhsuk-u-visually-hidden"
      >
        Error: 
      </span>
      Error message goes here
    </p>
    <select
      aria-describedby="select_:r4:--error-message"
      aria-labelledby="select_:r4:--label"
      as="select"
      class="nhsuk-select nhsuk-select--error"
      id="select_:r4:"
    >
      <option
        value="option-1"
      >
        Option 1
      </option>
      <option
        value="option-2"
      >
        Option 2
      </option>
      <option
        value="option-3"
      >
        Option 3
      </option>
    </select>
  </div>
</div>
`;

exports[`should render a select element with hint 1`] = `
<div>
  <div
    class="nhsuk-form-group"
  >
    <label
      class="nhsuk-label"
      for="select_:r2:"
      id="select_:r2:--label"
    >
      Label text goes here
    </label>
    <div
      class="nhsuk-hint"
      id="select_:r2:_:r3:--hint"
    >
      Hint text goes here
    </div>
    <select
      aria-describedby="select_:r2:_:r3:--hint"
      aria-labelledby="select_:r2:--label"
      as="select"
      class="nhsuk-select"
      id="select_:r2:"
    >
      <option
        value="option-1"
      >
        Option 1
      </option>
      <option
        value="option-2"
      >
        Option 2
      </option>
      <option
        value="option-3"
      >
        Option 3
      </option>
    </select>
  </div>
</div>
`;
